######################å¦‚æœæ²¡æœ‰gafæ–‡ä»¶
å¯ä»¥ç…§ç€ä»¥ä¸‹æ“ä½œåœ¨BioMartä¸‹è½½

âœ… æ–¹æ³•äºŒï¼šä½¿ç”¨ Ensembl çš„æ³¨é‡Šï¼ˆå¦‚ GTF + homology-based GOï¼‰
Ensembl å¯¹è£¸é¼¹é¼ æä¾›äº†åŸºå› ç»„æ³¨é‡Šï¼ˆç‰ˆæœ¬å¦‚ Heterocephalus_glaber_hg38ï¼‰ã€‚å¯ä»¥é€šè¿‡ Ensembl BioMart è·å–GOæ³¨é‡Šï¼š

ğŸ“ Ensembl BioMart æ“ä½œæ­¥éª¤ï¼š
æ‰“å¼€ï¼šhttps://www.ensembl.org/biomart/martview/

é€‰æ‹©æ•°æ®åº“ï¼šEnsembl Genes

é€‰æ‹©ç‰©ç§ï¼šHeterocephalus glaber

ç‚¹å‡»ã€Attributesã€‘ï¼Œé€‰æ‹©ï¼š

Gene stable ID

GO Term Accession

GO Term Name

ç‚¹å‡»ã€Resultsã€‘å¯¼å‡ºä¸º TSV/CSV æ–‡ä»¶

#####é€šè¿‡gafæ–‡ä»¶è¿›è¡Œéæ¨¡å¼ç‰©ç§çš„GSEA
library(GO.db)
library(clusterProfiler)

goterms <- Term(GOTERM)
term2name <- data.frame("GOID"=names(goterms),"term"=goterms )
head(term2name)

gaf_file <- "/home/mnt1/wanghao/project/Whale_IPS/IPS/IPS_PCA/pluripotency/GAF/GCF_002837175.3-RS_2023_04_gene_ontology.gaf"
gaf_data <- read.delim(gaf_file, header = FALSE, comment.char = "!", stringsAsFactors = FALSE)

#########è¿‡æ»¤å‡ºç”Ÿç‰©å­¦è¿‡ç¨‹ P (biological process), F (molecular function) or C (cellular component) çš„å“ªç§GOæ³¨é‡Š
gaf_data <- gaf_data   %>% 
    filter(V9 == "P")

gene_go_annotations <- gaf_data[, c(5, 3)]
colnames(gene_go_annotations) <- c("GOID","GeneSYMBOL")

library(GSEABase)

sperm <- read.table("/home/mnt1/wanghao/project/Whale_IPS/IPS/IPS_PCA/file_out/spermDevelop_DESeq2.out")  %>% 
          dplyr::select(log2FoldChange_EB10_vs_wipsc,padj_EB10_vs_wipsc,log2FoldChange_EB20_vs_EB10,padj_EB20_vs_EB10)
spermprotein <- read.table("/home/mnt1/wanghao/project/Whale_IPS/IPS/IPS_PCA/pluripotency/gtf/sperm_protein_coding_gene_names.txt")

sperm <- sperm[which(rownames(sperm) %in% spermprotein$V1),]


sperm <- sperm  %>% 
  arrange(-log2FoldChange_EB20_vs_EB10)

# å‡è®¾ä½ çš„æ•°æ®æ¡†æ˜¯sperm
sperm <- subset(sperm, !is.na(log2FoldChange_EB20_vs_EB10))


genelist <-setNames(sperm$log2FoldChange_EB20_vs_EB10,rownames(sperm))

egmt <- GSEA(genelist,TERM2GENE = gene_go_annotations,TERM2NAME = term2name ,minGSSize = 1)
